src/org/traccar/protocol/Tk103ProtocolEncoder.java

```diff
@@ -16,36 +16,89 @@
  */
 package org.traccar.protocol;
 
+import org.traccar.Context;
 import org.traccar.StringProtocolEncoder;
 import org.traccar.helper.Log;
 import org.traccar.model.Command;
 
 public class Tk103ProtocolEncoder extends StringProtocolEncoder {
 
+    protected String t580WFormatCommand(Command command, String format, String... keys) {
+        String content = super.formatCommand(command, format, keys);
+        return String.format("[begin]sms2,%s,[end]", content);
+    }
+
+    protected String checkEn(Command command, String ifEnabled, String ifDisabled) {
```

Comment:
Method and parameter names are not per guidelines. Also, I think it should be inlined.

Reply:
done